client.go

```diff
@@ -355,6 +355,9 @@ type RemoteContext struct {
 	// MaxConcurrentDownloads is the max concurrent content downloads for each pull.
 	MaxConcurrentDownloads int
 
+	// MaxConcurrentUploads is the max concurrent uploads for each push.
+	MaxConcurrentUploads int
```

Comment:
I see that this parallels the existing `MaxConcurrentDownloads`, but I wonder if it might be better to name it `MaxConcurrentUploadedLayers`.  Some resolvers (like [this one](https://github.com/awslabs/amazon-ecr-containerd-resolver/)) make concurrent requests for parts within a layer rather than a single request per layer.

Reply:
Agree with `MaxConcurrentUploadedLayers`. More accurate. Updated it later. Thanks!